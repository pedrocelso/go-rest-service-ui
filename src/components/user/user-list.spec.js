import React from 'react';
import {shallow} from 'enzyme'

import UserList from './user-list'

describe(`<UserList />`, () => {
  it(`Should fetch the users from service and render`, () => {
    const service = {
      getUsers: (_) => ({
        fork: (rej, res) => {
          res([
            { name: `john`, email: `john@whatever.com` },
            { name: `john t`, email: `john@whatever.com` },
            { name: `Renato Russo`, email: `renato@legiao.com.br` },
          ])
        }
      })
    }

    const component = shallow(
      <UserList service={service}/>,
    );

    expect(component).toMatchSnapshot();
  });

  it(`Should not display users if service returns an error`, () => {
    // const service = {
    //   getUsers: () => ({
    //     pipe: (f) => {
    //       console.log('================f================')
    //       console.log(f)
    //       console.log('==============================')
    //     }
    //   })
    // }
    const service = {
      getUsers: () => {
        console.log('================ENTROU================')
        return {
          pipe: (f) => {
            console.log('================f================')
            console.log(f)
            console.log('==============================')
          }
        }
      }
    }
    
    const component = shallow(
      <UserList service={service}/>,
    );

    expect(component).toMatchSnapshot();
  });
});