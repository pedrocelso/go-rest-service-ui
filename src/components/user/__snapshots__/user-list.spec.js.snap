// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserList /> Should fetch the users from service and render 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="list-group"
  >
    <div
      className="list-group-item list-group-item-action cursor-pointer"
      key="john@whatever.com"
    >
      john
       - 
      john@whatever.com
    </div>
    <div
      className="list-group-item list-group-item-action cursor-pointer"
      key="john@whatever.com"
    >
      john t
       - 
      john@whatever.com
    </div>
    <div
      className="list-group-item list-group-item-action cursor-pointer"
      key="renato@legiao.com.br"
    >
      Renato Russo
       - 
      renato@legiao.com.br
    </div>
  </div>
</div>
`;

exports[`<UserList /> Should not display users if service returns an error 1`] = `
<div
  className="panel panel-default"
/>
`;
